exploration/0475-overriding-extending-namespacing.md

```diff
@@ -0,0 +1,334 @@
+> functionality, while a Node application might use `import`.
+
+If an implementation supports user-installed formatters, selectors, function options,
+or expression annotations, it must also support providing "namespace" prefixes for
+each installed set of functionality.
+
+In this design, each namespace prefix is a short string and is associated with a URL.
+The URL is intended to point to some publically-available copy of the add-on library's
+registry description, for use by tooling and as a reference to users such as translators.
+
+There is no requirement that an implementation read the document at the end of the URL,
+nor validate the contents in general or against the actual functionality installed.
+
+> [!NOTE]
+> It is a good idea to use ASCII strings for namespace identifiers.
+> Remember that translators (and others) in many different languages and
+> with many different keyboards need to be able to enter the prefix.
+
+The namespace prefix is part of the `name` production.
+The prefix must be at least one character in length.
+It may be as long as desired, although users are cautioned that brevity
+is desirable.
+The prefix is separated from the name by a colon (U+003A COLON).
+
+The choice of a `:` is intentional, as it already used for function identification
+and might be familiar from similar usage in XML namespaces
+as well as slightly similar to C++, e.g. `ns::function`.
+This design leverages these sorts of "application familiarity"
+as well as the current syntax's use of colon as the function sigil.
+
+The namespace prefix `std` is reserved and refers to the default registry.
+The default registry will have a well-known URL under `unicode.org`
+but this URL is not yet specified.
+
+```abnf
+name      = [namespace] name-body
+namespace = name-start *name-char namespace-sep
+namespace-sep = ":"
+name-body = name-start *name-char
+```
+
+> [!NOTE]
+> The `name-start` and `name-char` productions will have to be altered to
+> **_not_** permit U+003A COLON in a name and to otherwise address
+> naming concerns.
+> This design document does not show the naming changes because there are
+> other issues in play for these names.
+> For now, just consider that `name-char` will have no colon.
+
+The `name` production as defined here applies to:
+
+- function (selector/formatting) names
+- option names
+- spanable names
+- expression annotation names (if approved)
+
+Examples:
+
+> Add-on function:
+>
+> ```
+> Today is {$today :icu:datetime skeleton=EEEEyMdjm}
+> ```
+>
+> Add on option:
+>
+> ```
+> Today is {$today :datetime icu:skeleton=EEEEyMdjm}
+> ```
+>
+> Add-on spannables (such as markup):
+>
+> ```
+> Today is {+html:a}{$today}{-html:a}
+> ```
+>
+> Add-on expression annotation:
+>
+> ```
+> Today is {$today :datetime @my:annotation}
+> ```
+>
+> Everything altogether all at once. This probably does not work
+> correctly, since `std:datetime` may not understand `icu:skeleton`:
```

Comment:
So can the example use `icu:datetime` instead?

Reply:
Good callout. I made a different change: I made this example use a built-in option and then did `icu:skeleton` for options after.